### PR TITLE
5.01 AST #4 - Overheal and CI 

### PR DESCRIPTION
**Pull request by suShirow**
Adapted from whm overheal and benison

### CODE REVIEW

**Comment by Fank** [comment]
:+1: 
image

**Comment by Fank** [comment]
@suShirow conflicts :eyes:

File: src/parser/jobs/ast/modules/CelestialIntersection.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		this.addHook('cast', {abilityId: ACTIONS.CELESTIAL_INTERSECTION.id, by: 'player'}, this.onCast)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private onCast(event: CastEvent) {
+		this.uses++
+		if (this.lastUse === 0) { this.lastUse = this.parser.fight.start_time }
+
+		const held = event.timestamp - this.lastUse - (ACTIONS.CELESTIAL_INTERSECTION.cooldown * 1000)
+		if (held > 0) {
```
Comment (ackwell) [inline src/parser/jobs/ast/modules/CelestialIntersection.tsx]: ubernit; could simplify this with just `this.totalHeld += Math.max(0, held)`

File: src/parser/jobs/ast/modules/CelestialIntersection.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			this.totalHeld += held
+		}
+		// update the last use
+		this.lastUse = event.timestamp
+	}
+
+	onComplete() {
+		const holdDuration = this.uses === 0 ? this.parser.fightDuration : this.totalHeld
+		const usesMissed = Math.floor(holdDuration / (ACTIONS.CELESTIAL_INTERSECTION.cooldown * 1000))
```
Comment (ackwell) [inline src/parser/jobs/ast/modules/CelestialIntersection.tsx]: might be worth factoring the multiplied cd out somewhere, it's being used in a fair few places

**Pull request merged**